src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java

```diff
@@ -392,29 +392,69 @@ public int hashCode() {
          * @return true if the source of event matches the text of this tag.
          */
         public boolean isMatch(TreeWalkerAuditEvent event) {
-            final int line = event.getLine();
             boolean match = false;
-
-            if (line >= firstLine && line <= lastLine) {
-                final Matcher tagMatcher = tagCheckRegexp.matcher(event.getSourceName());
-
-                if (tagMatcher.find()) {
-                    match = true;
-                }
-                else if (tagMessageRegexp == null) {
-                    if (event.getModuleId() != null) {
-                        final Matcher idMatcher = tagCheckRegexp.matcher(event.getModuleId());
-                        match = idMatcher.find();
-                    }
+            if (isInScopeOfSuppression(event)) {
+                if (isCheckMatch(event)) {
```

Comment:
in a future PR, this could be
```java
match = isCheckMatch(event) && isMessageMatch(event) && isIdMatch(event);
```
if there is an `idFormat` separate from `checkFormat` (following the example of `SuppressionFilter`)

Reply:
It could be `match = isInScopeOfSuppression(event) && isMessageMatch(event) (isCheckMatch(event) || isIdMatch(event));`